### PR TITLE
#1349 - Option to add description to crash dialog

### PR DESCRIPTION
**Pull request by T-0pel**
#1349 

This PR adds the option for the user to provide additional information about steps taken before a crash occurred. This description is then added as a text file to the diagnostic package.

There is another CrashWindow in the FullscreenApp project. Is that still work in progress? Should I add the textbox there as well?

### CODE REVIEW

File: source/Playnite/Localization/LocSource.xaml [hunk 1]
```diff
@@ -38,13 +38,14 @@ Indeterminate - No changes (when editing multiple games)</sys:String>
...
         
-If you want to help us fix this issue, please create diagnostics package and use 'Report Crash' button to let us know what happened. Thank you.</sys:String>
+If you want to help us fix this issue, please briefly describe actions taken before the crash, create a diagnostics package, and use the 'Report Crash' button to let us know what happened. Thank you.</sys:String>
     <sys:String x:Key="LOCCrashDescriptionFullscreen" xml:space="preserve">Unrecoverable error occurred.
         
 If you want to help us fix this issue, please send diagnostic information. Thank you.</sys:String>
     <sys:String x:Key="LOCCrashReportIssue">Report Crash</sys:String>
     <sys:String x:Key="LOCCrashRestartPlaynite">Restart Playnite</sys:String>
     <sys:String x:Key="LOCCrashClosePlaynite">Exit Playnite</sys:String>
+    <sys:String x:Key="LOCCrashUserActionsDescription">Actions taken before the crash:</sys:String>
```
Comment (JosefNemec) [inline source/Playnite/Localization/LocSource.xaml]: It should clarify that the text must be in English.

File: source/Playnite/Localization/LocSource.xaml [hunk 2]
```diff
@@ -38,13 +38,14 @@ Indeterminate - No changes (when editing multiple games)</sys:String>
     <sys:String x:Key="LOCCrashWindowTitle">Oh no, something bad happened…</sys:String>
     <sys:String x:Key="LOCCrashDescription" xml:space="preserve">Unrecoverable error occurred.
         
-If you want to help us fix this issue, please create diagnostics package and use 'Report Crash' button to let us know what happened. Thank you.</sys:String>
+If you want to help us fix this issue, please briefly describe actions taken before the crash, create a diagnostics package, and use the 'Report Crash' button to let us know what happened. Thank you.</sys:String>
```
Comment (JosefNemec) [inline source/Playnite/Localization/LocSource.xaml]: I think this whole string needs to be changed completely. If the user provides description of crash then it seems unnecessary to ask them to open github issue as well, where they will be asked to provide description again.

File: source/Playnite.DesktopApp/Windows/CrashHandlerWindow.xaml [hunk 3]
```diff
@@ -24,8 +24,12 @@
                     Command="{Binding ReportIssueCommand}"/>
         </DockPanel>
         <TextBlock Text="{DynamicResource LOCCrashDescription}"
-                   Margin="10,10,10,20"
+                   Margin="10,10,10,10"
                    VerticalAlignment="Center" DockPanel.Dock="Top" 
                    TextWrapping="Wrap" Foreground="{StaticResource TextBrush}" />
+        <Label Content="{DynamicResource LOCCrashUserActionsDescription}" DockPanel.Dock="Top"
+               Margin="10,10,10,0" FontSize="12"/>
+        <TextBox Name="TextBoxDescription" Text="{Binding Description}" Margin="10,10,10,10"
```
[the remark this answers is not in the document]
Comment (T-0pel) [reply source/Playnite.DesktopApp/Windows/CrashHandlerWindow.xaml]: Done.

File: source/Playnite/Localization/LocSource.xaml (hunk 1 above)
Comment (T-0pel) [reply source/Playnite/Localization/LocSource.xaml]: Done.

File: source/Playnite/Localization/LocSource.xaml (hunk 2 above)
Comment (T-0pel) [reply source/Playnite/Localization/LocSource.xaml]: I have modified the message. Please take a look and let me know what you think.

**Comment by JosefNemec** [comment]
I would maybe prefer this:
>If you want to help us fix this issue, please briefly describe actions taken before the crash and then create a diagnostics package. If you are online, the package will get uploaded to the Playnite server for analysis.
>
> Alternatively, you can use the 'Report Crash' button to create a new issue on GitHub and report the crash manually.

What do you think?

**Comment by T-0pel** [comment]
> I would maybe prefer this:
> 
> > If you want to help us fix this issue, please briefly describe actions taken before the crash and then create a diagnostics package. If you are online, the package will get uploaded to the Playnite server for analysis.
> > Alternatively, you can use the 'Report Crash' button to create a new issue on GitHub and report the crash manually.
> 
> What do you think?

Sounds good, I will modify and commit it today.

**Comment by T-0pel** [comment]
I have modified the text, please let me know if there is anything else you want me to change before merging it down.

**Pull request merged**